src/Rfc7230.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace GuzzleHttp\Psr7;
+
+final class Rfc7230
+{
+	/**
+	 * Header related regular expressions (copied from amphp/http package)
+	 * (Note: once we require PHP 7.x we could just depend on the upstream package)
+	 *
+	 * @link https://github.com/amphp/http/blob/16e465fa82555104d1cff98cb8e412295a380214/src/Rfc7230.php#L12-L15
+	 * @license https://github.com/amphp/http/blob/16e465fa82555104d1cff98cb8e412295a380214/LICENSE
+	 */
+	const HEADER_REGEX = "(^([^()<>@,;:\\\"/[\]?={}\x01-\x20\x7F]++):[ \t]*+((?:[ \t]*+[\x21-\x7E\x80-\xFF]++)*+)[ \t]*+\r\n)m";
```

Comment:
If you want to keep `\r` optional, then a `?` should be added after `\r`.

Reply:
I believe it was kept optional because the body was split as well. Since it's not anymore, I don't see why it should be optional. It's not even valid, is it?

Reply:
The specification doesn't allow them, correct, but historically many implementations made them optional, which makes things work with e.g. `netcat`.  We moved to making them required, but I think Apache also accepts only `\n`.

Reply:
Hm, I guess this is true for the HEADER_FOLD_REGEX as well.